src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1473,9 +1474,9 @@ private EvaluationContext getPotentiallyLimitedScopeContext(EvaluationContext qu
         return newReferences;
     }
 
-    private SelectChoice buildSelectChoice(TreeReference choiceRef, ItemsetBinding itemset,
-                                           DataInstance formInstance, EvaluationContext ec,
-                                           ReducingTraceReporter reporter, int index) {
+    public static SelectChoice buildSelectChoice(TreeReference choiceRef, ItemsetBinding itemset,
```

Comment:
I think we should definitely move this into a helper class to make it clearer how its context is built and protect potential scope issues (you could imagine an accidental static reference within this file being made to an element in the method), and make it clear that this isn't just an XForm operation